artemis/modules/data/offat/offat_artemis.patch

```diff
@@ -0,0 +1,224 @@
+             "' AND SLEEP(5) --",
+         ]
+
++        SQL_INJECTION_TIME_THRESHOLD = 5
++
++        sql_injection_sleep_payloads = [
++            f"sleep({SQL_INJECTION_TIME_THRESHOLD})",
++            f"pg_sleep({SQL_INJECTION_TIME_THRESHOLD})",
++            f"'||sleep({SQL_INJECTION_TIME_THRESHOLD})||'",
++            f"'||pg_sleep({SQL_INJECTION_TIME_THRESHOLD})||'",
++        ]
++
+         tasks = []
+         for sqli_payload in basic_sqli_payloads:
+             for path_obj in endpoints_with_param_in_path:
+@@ -409,6 +429,7 @@ class TestGenerator:
+                             False: 'Endpoint is not vulnerable to SQLi',
+                         },
+                         'success_codes': success_codes,
++                        'sleep_payload': sqli_payload in sql_injection_sleep_payloads,
+                         'response_filter': PostTestFiltersEnum.STATUS_CODE_FILTER.name,
+                     }
+                 )
+diff --git a/src/offat/tester/handler.py b/src/offat/tester/handler.py
+index 44bf376..e8663bb 100644
+--- a/src/offat/tester/handler.py
++++ b/src/offat/tester/handler.py
+@@ -28,6 +28,7 @@ def generate_and_run_tests(
+     capture_failed: bool = False,
+     remove_unused_data: bool = True,
+     ssl_verify: bool = True,
++    only_get_requests: bool = False,
+ ):
+     """
+     Generates and runs tests for the provided OAS/Swagger file.
+@@ -70,6 +71,7 @@ def generate_and_run_tests(
+         headers=req_headers,
+         proxies=proxies,
+         ssl_verify=ssl_verify,
++        only_get_requests=only_get_requests,
+     )
+
+     results: list = []
+diff --git a/src/offat/tester/post_test_processor.py b/src/offat/tester/post_test_processor.py
+index de0d812..cf6f028 100644
+--- a/src/offat/tester/post_test_processor.py
++++ b/src/offat/tester/post_test_processor.py
+@@ -138,6 +138,39 @@ class PostRunTests:
+             new_results.append(new_result)
+
+         return new_results
++
++    @staticmethod
++    def check_time_based_sqli(results: list[dict], SQL_INJECTION_SLEEP_THRESHOLD: float = 5) -> list[dict]:
++        """
++        Checks SQLi results for time-based blind SQL injection vulnerabilities.
++
++        Args:
++            results (list[dict]): List of test results.
++            SQL_INJECTION_SLEEP_THRESHOLD (float): Time threshold (in seconds) to determine a time-based SQLi.
++
++        Returns:
++            list[dict]: Updated list with vulnerability marked for time-based SQLi.
++        """
++        updated_results = []
++
++        for result in results:
++            test_name = result.get("test_name", "")
++            is_sql_test = "SQLi" in test_name or "sql" in test_name.lower()
++            sleep_payload = result.get("sleep_payload", False)
++            response_time = result.get("response_time")
++
++            new_result = deepcopy(result)
++
++            if is_sql_test and sleep_payload and isinstance(response_time, (int, float)):
```

Comment:
I'm afraid this is not correct - this will not perform this check if it's not a sleep payload, so when entering e.g. `'` as the payload, we will still treat error 500 as a sign of vulnerability. Can we change this to have only sleep payloads?